jax/lax/lax.py

```diff
@@ -1164,7 +1164,7 @@ def index_in_dim(operand, index, axis=0, keepdims=True):
 
 def dynamic_slice_in_dim(operand, start_index, slice_size, axis=0):
   """Convenience wrapper around dynamic_slice applying to one dimension."""
-  start_indices = [onp.array([0])] * operand.ndim
+  start_indices = [onp.array([0], dtype=start_index.dtype)] * operand.ndim
```

Comment:
Maybe start_index could be a Python scalar (i.e. a static slice even though we're calling `dynamic_slice_in_dim`) and so we'd be better off using `_dtype(start_index)` here.

Reply:
Done.